### PR TITLE
CLN: changed .format to f-string in pandas/core/indexes

### PR DESCRIPTION
**Pull request by makeajourney**
towards #29547

- [ ] closes  
- [x] tests passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -432,10 +432,8 @@ def closed(self):
     )
     def set_closed(self, closed):
         if closed not in _VALID_CLOSED:
-            msg = "invalid option for 'closed': {closed}"
-            raise ValueError(msg.format(closed=closed))
+            raise ValueError(f"invalid option for 'closed': {closed}")
 
-        # return self._shallow_copy(closed=closed)
```
Comment (simonjayhawkins) [inline pandas/core/indexes/interval.py]: is there a reason to remove this comment?

File: pandas/core/indexes/period.py [hunk 2]
```diff
@@ -183,9 +183,8 @@ def __new__(
 
         if not set(fields).issubset(valid_field_set):
             raise TypeError(
-                "__new__() got an unexpected keyword argument {}".format(
-                    list(set(fields) - valid_field_set)[0]
-                )
+                "__new__() got an unexpected keyword argument "
+                f"{list(set(fields) - valid_field_set)[0]}"
```
Comment (simonjayhawkins) [inline pandas/core/indexes/period.py]: could do this outside f-space.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @makeajourney for the PR. generally lgtm.

File: pandas/core/indexes/interval.py (hunk 1 above)
Comment (makeajourney) [reply pandas/core/indexes/interval.py]: No. I put it back as it was placed.

File: pandas/core/indexes/period.py (hunk 2 above)
Comment (makeajourney) [reply pandas/core/indexes/period.py]: separated that line with the new variable.

**Review by topper-123 — APPROVED** [review]

**Comment by topper-123** [comment]
Thanks, @makeajourney . Welcome to take more of these.

**Pull request merged**